webapp/app/js/perf.js

```diff
@@ -28,3 +41,207 @@ perf.factory('getSeriesSummary', [ function() {
              options: options, subtestSignatures: subtestSignatures };
   };
 }]);
+
+
+perf.factory('getCounterMap', [ 'isReverseTest',
+  function(isReverseTest) {
+    return function(testName, originalData, newData) {
+      var cmap = {'originalGeoMean': NaN, 'originalRuns': 0, 'originalStddev': NaN,
+                  'newGeoMean': NaN, 'newRuns': 0, 'newStddev': NaN,
+                  'delta': NaN, 'deltaPercentage': NaN, 'isEmpty': false,
+                  'isRegression': false, 'isImprovement': false, 'isMinor': true};
+
+      if (originalData) {
+         cmap.originalGeoMean = originalData.geomean;
+         cmap.originalRuns = originalData.runs;
+         cmap.originalStddev = originalData.stddev;
+         cmap.originalStddevPct = ((originalData.stddev / originalData.geomean) * 100).toFixed(2);
+      }
+      if (newData) {
+         cmap.newGeoMean = newData.geomean;
+         cmap.newRuns = newData.runs;
+         cmap.newStddev = newData.stddev;
+         cmap.newStddevPct = ((newData.stddev / newData.geomean) * 100).toFixed(2);
+      }
+
+      if ((cmap.originalRuns == 0 && cmap.newRuns == 0) ||
+          (testName == 'tp5n summary opt')) {
+        // We don't generate numbers for tp5n, just counters
+        cmap.isEmpty = true;
+      } else {
+        cmap.delta = (cmap.newGeoMean - cmap.originalGeoMean).toFixed(2);
+        cmap.deltaPercentage = (cmap.delta / cmap.originalGeoMean * 100).toFixed(2);
+        if (cmap.deltaPercentage > 2.0) {
+          cmap.isMinor = false;
+          isReverseTest(testName) ? cmap.isImprovement = true : cmap.isRegression = true;
+        } else if (cmap.deltaPercentage < -2.0) {
+          cmap.isMinor = false;
+          isReverseTest(testName) ? cmap.isRegression = true : cmap.isImprovement = true;
+        }
+      }
+      return cmap;
+    }
+}]);
+
+perf.factory('calculateStats', [ 'math', function(math) {
+  return function(perfData, resultSetID) {
+    var geomeans = [];
+    var total = 0;
+    _.where(perfData, { result_set_id: resultSetID }).forEach(function(pdata) {
+      //summary series have geomean, individual pages have mean
+      if (pdata.geomean === undefined) {
+        geomeans.push(pdata.mean);
+      } else {
+        geomeans.push(pdata.geomean);
+      }
+    });
+
+    geomeans.forEach(function(mean) { total += mean; });
```

Comment:
You could express this slightly more concisely as:

    var total = _.reduce(means, function(mean, total) { return total + mean; })